_stbt/core.py

```diff
@@ -1531,12 +1545,13 @@ def push_sample(self, sample):
 
     def on_error(self, pipeline, _bus, message):
         assert message.type == Gst.MessageType.ERROR
-        Gst.debug_bin_to_dot_file_with_ts(
-            pipeline, Gst.DebugGraphDetails.ALL, "ERROR")
+        if pipeline is not None:
+            # If we receive an error after the pipeline has been torn down:
+            Gst.debug_bin_to_dot_file_with_ts(
+                pipeline, Gst.DebugGraphDetails.ALL, "ERROR")
         err, dbg = message.parse_error()
         self.tell_user_thread(
             UITestError("%s: %s\n%s\n" % (err, err.message, dbg)))
-        _mainloop.quit()
```

Comment:
Hmmm, I suppose if we had previously received an error then when we emit end-of-stream during teardown we won't take the FlowReturn.OK branch and therefore teardown won't actually take 10s?